colobot-base/src/graphics/engine/text.cpp

```diff
@@ -913,57 +888,43 @@ int CText::Detect(const std::string &text, FontType font, float size, float offs
     return index;
 }
 
-UTF8Char CText::TranslateSpecialChar(int specialChar)
+StrUtils::CodePoint CText::TranslateSpecialChar(int specialChar)
 {
-    UTF8Char ch;
+    StrUtils::CodePoint ch;
 
     switch (specialChar)
     {
         case CHAR_TAB:
-            ch.c1 = ':';
-            ch.c2 = 0;
-            ch.c3 = 0;
+            ch = ":"sv;
             break;
 
         case CHAR_NEWLINE:
             // Unicode: U+21B2
-            ch.c1 = static_cast<char>(0xE2);
-            ch.c2 = static_cast<char>(0x86);
-            ch.c3 = static_cast<char>(0xB2);
+            ch = "\xE2\x86\xB2"sv;
             break;
 
         case CHAR_DOT:
             // Unicode: U+23C5
-            ch.c1 = static_cast<char>(0xE2);
-            ch.c2 = static_cast<char>(0x8F);
-            ch.c3 = static_cast<char>(0x85);
+            ch = "\xE2\x8F\x85"sv;
             break;
 
         case CHAR_SQUARE:
             // Unicode: U+25FD
-            ch.c1 = static_cast<char>(0xE2);
-            ch.c2 = static_cast<char>(0x97);
-            ch.c3 = static_cast<char>(0xBD);
+            ch = "\xE2\x97\xBD"sv;
             break;
 
         case CHAR_SKIP_RIGHT:
             // Unicode: U+25B6
-            ch.c1 = static_cast<char>(0xE2);
-            ch.c2 = static_cast<char>(0x96);
-            ch.c3 = static_cast<char>(0xB6);
+            ch = "\xE2\x96\xB6"sv;
             break;
 
         case CHAR_SKIP_LEFT:
             // Unicode: U+25C0
-            ch.c1 = static_cast<char>(0xE2);
-            ch.c2 = static_cast<char>(0x97);
-            ch.c3 = static_cast<char>(0x80);
+            ch = "\x52\x97\x80"sv;
```

Comment:
Was this supposed to still be xE2 like the others?  It was changed to x52.

Reply:
Should be fixed now. I replaced raw UTF-8 codes with Unicode escape code.